lib/k8s/util.rb

```diff
@@ -25,6 +25,20 @@ def self.compact_map(args)
       args.map{ |arg| value_map[arg] }
     end
 
+    # Recursive compact for Hash/Array
+    #
+    # @param hash_or_array [Hash,Array]
+    # @return [Hash,Array]
+    def self.recursive_compact(hash_or_array)
+      p = proc do |*args|
+        v = args.last
+        v.delete_if(&p) if v.respond_to?(:delete_if)
+        v.nil? || v.respond_to?(:"empty?") && v.empty?
```

Comment:
```suggestion
        v.nil? || (v.respond_to?(:empty?) && v.empty?)
```

no need for quotes, parenthesis added for clarity.